torch/multiprocessing/spawn.py

```diff
@@ -209,8 +209,9 @@ def join(
                     exit_code=exitcode,
                 )
 
-        with open(self.error_files[error_index], "rb") as fh:
+        with open(self.__error_files[error_index], "rb") as fh:
```

Comment:
Proper way to handle this is an contextlib.ExitStack with an unlink callback. Then it will be properly deleted on exceptions too.

@cyyever this is the proper way to handle your other recent tmpfile changes that call os.remove.